src/linear_algebra/cross_product.lean

```diff
@@ -0,0 +1,158 @@
+/-
+Copyright (c) 2021 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak, Kyle Miller, Eric Wieser
+-/
+
+import data.matrix.notation
+import linear_algebra.bilinear_map
+import linear_algebra.matrix.determinant
+import algebra.lie.basic
+
+/-!
+# Cross products
+
+This module defines the cross product of vectors in $R^3$ for $R$ a commutative ring.
+
+## Main definitions
+
+* `cross_product` is the cross product of pairs of vectors in $R^3$.
+
+## Main results
+
+* `triple_product_eq_det`
+* `cross_dot_cross`
+* `jacobi_identity`
+
+## Notation
+
+The locale `matrix` gives the following notation:
+
+* `×₃` for the cross product
+
+## Tags
+
+crossproduct
+-/
+
+open matrix
+open_locale matrix
+
+variables {R : Type*} [comm_ring R]
+
+/-- The cross product of two vectors in $R^3$ for $R$ a commutative ring. -/
+def cross_product : (fin 3 → R) →ₗ[R] (fin 3 → R) →ₗ[R] (fin 3 → R) :=
+begin
+  apply linear_map.mk₂ R (λ (a b : fin 3 → R),
+    ![a 1 * b 2 - a 2 * b 1,
+      a 2 * b 0 - a 0 * b 2,
+      a 0 * b 1 - a 1 * b 0]);
+  intros;
+  simp only [@vec3_add R _ _ _ _ _ _ _,
+    pi.add_apply, algebra.id.smul_eq_mul, matrix.smul_cons, matrix.smul_empty, pi.smul_apply];
+  apply vec3_eq;
+  -- ring_nf is slow here
+  simp [add_comm, add_assoc, add_left_comm, mul_comm, mul_assoc, mul_left_comm,
+    add_mul, mul_add, sub_eq_add_neg],
+end
+
+localized "infixl ` ×₃ `: 74 := cross_product" in matrix
+
+lemma cross_product_apply (a b : fin 3 → R) :
+  a ×₃ b = ![a 1 * b 2 - a 2 * b 1,
+             a 2 * b 0 - a 0 * b 2,
+             a 0 * b 1 - a 1 * b 0] :=
+rfl
+
+section products_properties
+
+lemma cross_product_anticomm (v w : fin 3 → R) :
```

Comment:
Could you please create an `alias` for this called `neg_cross_product`?
Also, I think this should be tagged with `@[simp]`.

Reply:
If this is a `simp` lemma, then `cross_product_anticomm'` ought to be as well for confluence, I believe.